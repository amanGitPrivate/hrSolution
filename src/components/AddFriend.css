
:local(.addFriendSubmitButton) {
  border-radius: 0;
  border-color: #ABAAAA;
  border-left: 0;
  border-right: 0;
  display: inline-block;
}

:local(.addFriendSubmitButton) {
  margin-left: 5px;
  width: 50%;
  margin-top: 5px;
  vertical-align: top;
  text-align: center;
}


:local(.addFriendForm) {
  margin: 10px;
}

:local(.submitButton) {
  text-align: center;
}

