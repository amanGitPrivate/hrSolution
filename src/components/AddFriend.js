import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './AddFriend.css';
import AddFriendOption from './AddFriendOption';
import AddFriendInputBox from './AddFriendInputBox';

class AddFriendInput extends Component {
  constructor(props, context) {
    super(props, context);
    this.state = {
      name: this.props.name || '',
      gender: this.props.gender || 'noOptionSelected',
      validation: {
        name: false,
        gender: false
      }
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.triggerSubmitOnEnter = this.triggerSubmitOnEnter.bind(this);
  }

  handleChange(e) {
    const { name, value } = e.target;
    this.setState({ [name]: value });
    this.setState({
      validation: {
        ...this.state.validation,
        [name]: true
      }
    });
  }

  isFormValid() {
    const { name, gender } = this.state;
    let validation = {
      name: !!name,
      gender: gender !== 'noOptionSelected'
    };
    this.setState({ validation });

    if (!name || gender === 'noOptionSelected') {
      return false;
    }

    return true;
  }

  handleSubmit() {
    const { name, gender } = this.state;
    if (this.isFormValid()) {
      this.props.addFriend({ name, gender });
      this.setState({
        name: '',
        gender: 'noOptionSelected',
        validation: {
          name: false,
          gender: false
        }
      });
    }
  }

  triggerSubmitOnEnter(e) {
    if (e.keyCode === 13) {
      this.handleSubmit();
    }
  }

  render() {
    const { name, validation, gender } = this.state;
    return (
      <div className={styles.addFriendForm}>
        <AddFriendInputBox
          name={name}
          onChange={this.handleChange}
          triggerSubmitOnEnter={this.triggerSubmitOnEnter}
        />
        <AddFriendOption onChange={this.handleChange} gender={gender} />
        <div className={styles.submitButton}>
          <button
            className={classnames(
              'btn btn-default',
              styles.addFriendSubmitButton
            )}
            onClick={this.handleSubmit}
            disabled={!validation.name || !validation.gender}
          >
            Submit
          </button>
        </div>
      </div>
    );
  }
}

AddFriendInput.propTypes = {
  addFriend: PropTypes.func.isRequired
};

export default AddFriendInput;
