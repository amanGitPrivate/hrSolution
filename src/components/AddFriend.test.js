import React from 'react';
import AddFriend from './AddFriend';
import renderer from 'react-test-renderer';

const addFriend = jest.fn();

test('AddFriend renders correctly', () => {
  const component = renderer.create(<AddFriend addFriend={addFriend} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('AddFriend handleChange click test', () => {
  const component = renderer.create(<AddFriend addFriend={addFriend} />);
  const instance = component.root.instance;
  instance.handleChange({ target: { name: 'gender', value: 'male' } });
  console.assert(instance.state.gender === 'male');
});

test('AddFriend handleSubmit click test', () => {
  const component = renderer.create(<AddFriend addFriend={addFriend} />);
  const instance = component.root.instance;
  instance.state.gender = 'male';
  instance.state.name = 'aman';
  instance.handleSubmit();
});
