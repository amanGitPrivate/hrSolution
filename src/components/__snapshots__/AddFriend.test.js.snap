exports[`test AddFriend renders correctly 1`] = `
<div
  className={undefined}>
  <div>
    <input
      autoFocus={true}
      className="form-control"
      name="name"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Type the name of a friend"
      type="text"
      value="" />
  </div>
  <div
    className="form-control radio">
    <label
      className="">
      <input
        checked={false}
        name="gender"
        onChange={[Function]}
        type="radio"
        value="male" />
       
      male
    </label>
    <label>
      <input
        checked={false}
        name="gender"
        onChange={[Function]}
        type="radio"
        value="female" />
       
      female
       
    </label>
  </div>
  <div
    className={undefined}>
    <button
      className="btn btn-default"
      disabled={true}
      onClick={[Function]}>
      Submit
    </button>
  </div>
</div>
`;
